fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1946,21 +1948,25 @@ private Entry processEntryForGet(Entry.Request entryRequest, FHIRUrlParser reque
             if ("_search".equals(pathTokens[0])) {
                 resource = doSearch("Resource", null, null, queryParams, absoluteUri, null);
             } else {
+                checkResourceType(pathTokens[0]);
                 resource = doSearch(pathTokens[0], null, null, queryParams, absoluteUri, null);
             }
         } else if (pathTokens.length == 2) {
             // This is a 'read' request.
+            checkResourceType(pathTokens[0]);
             resource = doRead(pathTokens[0], pathTokens[1], true, false, null).getResource();
         } else if (pathTokens.length == 3) {
             if ("_history".equals(pathTokens[2])) {
                 // This is a 'history' request.
```

Comment:
Don't you need to check the resource type here?

Reply:
good catch.  i've added it now.